### PR TITLE
Refactor aggregate

### PR DESCRIPTION
**Pull request by OVI3D0**
Description
Refactor the aggregate feature to make individual functions more single-purpose, as well as renaming variables and adding more explanatory comments.

Testing
 New functionality includes testing
`make test`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: osbenchmark/aggregator.py [hunk 1]
```diff
@@ -22,37 +22,37 @@ def __init__(self, cfg, test_executions_dict, args):
...
-                task_name = task.name
-                task_name_iterations = f"{task_name}_iterations"
-                if task_name_iterations in workload_params:
-                    iterations = int(workload_params[task_name_iterations])
-                else:
-                    iterations = task.iterations or 1
-                self.accumulated_iterations[test_execution_id][task_name] = iterations
-        else:
-            raise ValueError(f"Test procedure '{self.test_procedure_name}' not found in the loaded workload.")
+        for task in self.loaded_workload.find_test_procedure_or_default(self.test_procedure_name).schedule:
```
Comment (IanHoang) [inline osbenchmark/aggregator.py:30]: Much more clean!

File: osbenchmark/aggregator.py [hunk 2]
```diff
@@ -147,8 +146,30 @@ def build_aggregated_results(self):
...
+        self.config.add(config.Scope.applicationOverride, "system", "time.start", current_timestamp)
+        self.config.add(config.Scope.applicationOverride, "test_execution", "pipeline", test_execution.pipeline)
+        self.config.add(config.Scope.applicationOverride, "workload", "params", test_execution.workload_params)
+        self.config.add(config.Scope.applicationOverride, "builder",
+                        "provision_config_instance.params", test_execution.provision_config_instance_params)
+        self.config.add(config.Scope.applicationOverride, "builder", "plugin.params", test_execution.plugin_params)
+        self.config.add(config.Scope.applicationOverride, "workload", "latency.percentiles", test_execution.latency_percentiles)
+        self.config.add(config.Scope.applicationOverride, "workload", "throughput.percentiles", test_execution.throughput_percentiles)
+
+    def build_aggregated_results(self):
```
Comment (IanHoang) [inline osbenchmark/aggregator.py:170]: We should add a return type hint

**Review by IanHoang — CHANGES_REQUESTED** [review]

File: osbenchmark/aggregator.py [hunk 3]
```diff
@@ -147,8 +146,30 @@ def build_aggregated_results(self):
 
             aggregated_results["op_metrics"].append(op_metric)
 
-        # extract the necessary data from the first test execution, since the configurations should be identical for all test executions
+        return aggregated_results
+
+    def update_config_object(self, test_execution):
```
[the remark this answers is not in the document]
Comment (OVI3D0) [reply osbenchmark/aggregator.py]: True, added type hints to all the functions now!

File: osbenchmark/aggregator.py [hunk 4]
```diff
@@ -22,37 +22,37 @@ def __init__(self, cfg, test_executions_dict, args):
...
 
     def aggregate_json_by_key(self, key_path: Union[str, List[str]]) -> Any:
+        """
+        Aggregates JSON results across multiple test executions using a specified key path.
+        Handles nested dictionary structures and calculates averages for numeric values
+        """
         all_jsons = [self.test_store.find_by_test_execution_id(id).results for id in self.test_executions.keys()]
 
-        # retrieve nested value from a dictionary given a key path
         def get_nested_value(obj: Dict[str, Any], path: List[str]) -> Any:
```
[the remark this answers is not in the document]
Comment (OVI3D0) [reply osbenchmark/aggregator.py]: You're right, renamed a bunch of variables in the aggregate_json_by_key function. Hopefully it's more clear now

File: osbenchmark/aggregator.py [hunk 5]
```diff
@@ -66,8 +66,7 @@ def aggregate_helper(objects: List[Any]) -> Any:
             if not objects:
```
[the remark this answers is not in the document]
Comment (OVI3D0) [reply osbenchmark/aggregator.py]: renamed + added comments now! lmk what you think

File: osbenchmark/aggregator.py (hunk 2 above)
Comment (OVI3D0) [reply osbenchmark/aggregator.py]: Done ✅

**Review by IanHoang — APPROVED** [review]
LGTM, thanks for making these changes

**Pull request merged**